gateway/src/api/latest.rs

```diff
@@ -362,6 +352,91 @@ async fn request_acme_certificate(
     Ok("certificate created".to_string())
 }
 
+#[instrument(skip_all, fields(%project_name, %fqdn))]
+async fn renew_acme_certificate(
+    _: Admin,
+    State(RouterState {
+        service, sender, ..
+    }): State<RouterState>,
+    Extension(acme_client): Extension<AcmeClient>,
+    Extension(resolver): Extension<Arc<GatewayCertResolver>>,
+    Path((project_name, fqdn)): Path<(ProjectName, String)>,
+    AxumJson(credentials): AxumJson<AccountCredentials<'_>>,
+) -> Result<String, Error> {
+    let fqdn: FQDN = fqdn
+        .parse()
+        .map_err(|_err| Error::from(ErrorKind::InvalidCustomDomain))?;
+
+    let account = AccountWrapper::from(credentials).0;
+    let fqdn_clone = fqdn.clone();
+    service
+        .new_task()
+        .project(project_name)
+        .and_then(task::run(move |ctx| {
+            let service_clone = service.clone();
+            let fqdn_clone_clone = fqdn_clone.clone();
+            let acme_client_clone = acme_client.clone();
+            let account_clone = account.clone();
+            let resolve_clone = resolver.clone();
+            async move {
+                // If project not ready yet, don't attept certificare renewal.
+                if !ctx.state.is_ready() {
+                    return TaskResult::Pending(ctx.state);
+                }
+
+                // Try retrieve the current certificate if any.
+                match service_clone
+                    .project_details_for_custom_domain(&fqdn_clone_clone)
+                    .await
+                {
+                    Ok(CustomDomain { certificate, .. }) => {
+                        let (_, x509_cert_chain) =
+                            parse_x509_certificate(certificate.as_bytes()).unwrap();
+                        let diff = x509_cert_chain
+                            .validity()
+                            .not_after
+                            .sub(ASN1Time::now())
+                            .unwrap();
+                        // If current certificate validity less_or_eq than 30 days, attempt
+                        // renewal.
+                        if diff.whole_days() <= 30 {
```

Comment:
If this is fired only through the admin command, why is there a check for expiration here? IOW, why don't we just expose the ability to (re)create certificates through the admin subcommand? And then whatever calls that command can do it on arriving close to expiration (and we gain the ability to renew them out of band should we want to).

For that, we could just expose an admin endpoint for listing custom certs maybe?

Reply:
For sure, keeping the automatic renewal out of the band should be cleaner. Then what is left is only the certificate creation, which is already contained by the `admin/acme/renew` API. The difference between this API and the `admin/acme/request` API is that the latter will additionally destroy and recreate the project. 

I will remove the task from the `admin/acme/renew` API and rely on an out-of-band automated renewal (cron job-like component for each project).

Reply:
Removed the automation for both gateway and custom domain certificates renewal. We'll have to rely on an out of band mechanism that calls into the shuttle-admin commands or gateway APIs. Let me know if you would like to change something or add more to this.